### PR TITLE
Configure codecov

### PR DESCRIPTION
**Pull request by shuheiktgw**
## Problem Statement

I'd like to try configuring Codecov again so that we can visualize and track test coverage over time.
See [the related Slack thread](https://kubernetes.slack.com/archives/C047LA9MUPJ/p1703655390563569) for the discussion.

## Related Issue

N/A

## Proposed Changes

Configure Codecov integration without blocking any PRs.

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+coverage:
```
Comment (shuheiktgw) [inline codecov.yml:1]: I'd like to set them as informational, meaning it won't block any PRs at least for now. 

https://docs.codecov.com/docs/quick-start#tips-and-tricks

File: codecov.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  status:
+    project:
+      default:
+        informational: true
+    patch:
+      default:
+        informational: true
+
+# When modifying this file, please validate using
+# curl -X POST --data-binary @codecov.yml https://codecov.io/validate
```
Comment (moolen) [inline codecov.yml:11]: Thanks for taking the lead on this one :heart:!

Can we add some [ignore paths](https://docs.codecov.com/docs/ignoring-paths)? E.g. `**/zz_generated.deepcopy.go`

Do you know if there's a way to make codecov to NOT add annotations to the file? If you scroll down in the [PR/files view](https://github.com/external-secrets/external-secrets/pull/2995/files).
It's also visible in the [GH check run](https://github.com/external-secrets/external-secrets/pull/2995/checks?check_run_id=20123445906).
The problem is, that this makes reviewing really tedious to scroll over the annotations :/ I'd prefer to NOT add any annotations. Found the [docs related to this here](https://docs.codecov.com/docs/github-checks#yaml-configuration-for-github-checks-and-codecov).

Comment (shuheiktgw) [reply codecov.yml]: Thanks for your feedback, @moolen!
I've turned off the annotations and configured ignore paths. Would you take a look at it again when you have time?

**Review by moolen — APPROVED** [review]
LGTM, thank you for this! :heart:

**Pull request merged**